src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,346 @@
+                "pushkin",
+                "ThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100Symbols",
+                "1"));
+        List<String> specialChars = getSpecialCharacters("");
+        specialChars.addAll(getRussianAndUkrLetters());
+
+        correctNames.addAll(specialChars);
+
+        return correctNames;
+    }
+
+    private static List<String> createWrongUserNames() {
+
+        return List.of(
+                "",
+                " ",
+                "aaa bbb",
+                " aaa",
+                "aaa ");
+    }
+
+    private static List<String> createCorrectWordsEmptyOrMin2Max50() {
+        List<String> correctWords = new ArrayList<>(List.of(
+                "aZ",
+                "99",
+                "Za0",
+                "wW1-",
+                "wW2-w",
+                "wW3'",
+                "wW4'w",
+                "wW5`",
+                "wW6`w",
+                "wW7_",
+                "wW8_w"
+        ));
+
+        List<String> tempList = new ArrayList<>();
+        correctWords.forEach(word -> getRussianAndUkrLetters().forEach(
+                letter -> tempList.add(word.concat(letter))
+        ));
+
+        correctWords.addAll(tempList);
+        correctWords.addAll(List.of(
+                "",
+                "LengthIs50LengthIs50LengthIs50LengthIs50LengthIs50"));
+
+        return correctWords;
+    }
+
+    private static List<String> createWrongWordsEmptyOrMin2Max50() {
+        String correctWord = "a1A";
+
+        List<String> wrongWords = new ArrayList<>(List.of(
+                "w",
+                "W",
+                "1",
+                " ",
+                " 1",
+                " W",
+                " w",
+                "1 ",
+                "1 ",
+                "W ",
+                "w ",
+                "ThisWordHasLengthMoreThan50CharactersItsLengthIs51S"));
+
+        List<String> specialChars = getSpecialCharacters("");
+        specialChars.removeAll(List.of("_", "-", "`", "'"));
+
+        specialChars.forEach(specialChar -> wrongWords.add(correctWord.concat(specialChar)));
+
+        return wrongWords;
+    }
+
+    private static List<String> getSpecialCharacters(String escapedSymbol) {
+        List<String> specialCharacters = new ArrayList<>();
+
+        IntStream.range(33, 48).forEach(i -> specialCharacters.add(escapedSymbol + ((char) i)));
+        IntStream.range(58, 65).forEach(i -> specialCharacters.add(escapedSymbol + ((char) i)));
+        IntStream.range(91, 97).forEach(i -> specialCharacters.add(escapedSymbol + ((char) i)));
+        IntStream.range(123, 127).forEach(i -> specialCharacters.add(escapedSymbol + ((char) i)));
+
+        return specialCharacters;
+    }
+
+    private static List<String> getRussianAndUkrLetters() {
+        List<String> russianAndUkrLetters = new ArrayList<>();
+
+        russianAndUkrLetters.add(String.valueOf((char) 1025));
```

Comment:
Such numbers look like 'magic numbers'. You should make them as constant or create a method that will explain what are you doing here.